Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -73,36 +71,89 @@ extension HTTPClient {
                 throw HTTPClientError.invalidURL
             }
 
-            try self.init(url: url, version: version, method: method, headers: headers, body: body)
+            self.init(url: url, version: version, method: method, headers: headers, body: body)
         }
 
-        public init(url: URL, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) throws {
-            guard let scheme = url.scheme else {
+        public init(url: URL, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) {
```

Comment:
I think the HTTP version should either not be passed here or should be an optional. Usually a user doesn't want to specify an HTTP version and you'd expect the client to choose the 'best available'. So I'd say, make it optional, default to `nil` and make it the last parameter.

Reply:
Internally you can _for now_ default to `1.1` but once we get HTTP/2.0 support in this client we'd likely prefer 2 and fall back to 1.1

Reply:
and I think that should be possible without breaking the API

Reply:
How will it differ from having it as a default parameter? What options do we have in choosing the version? Also, having it as non-optional would be a bit more convenient for user who wants to inspect the request for example, no?

Reply:
Or better question: what information will we use to fallback from HTTP 2 to 1.1?

Reply:
@artemredkin well, if you set it to `version = ...(major: 1, minor: 1)` then you can't change it to something else later. But if you change it to `nil` then you can change the interpretation of `nil` at any point in time. I'd probably rename the parameter to `forceHTTPVersion: HTTPVersion? = nil` to make it clear that the user is usually not expected to use it.

Usually, the user doesn't care at all what HTTP version is used (why would they?). Therefore I think the default should probably be: try HTTP 2 and if the upgrade didn't work, just use 1.1 instead. But we wouldn't want to expose this 'try 2, fall back to 1' to the user.

Also once we've got a connection pool we might want to do: Have a look in the connection pool, if there's something active already, use it, irregardless of what the HTTP version is. But if we need a new connection, then we do the normal 'try 2, fall back to 1' dance.

Does that make sense?

Reply:
What about removing version field altogether? Decision between 2 and 1.1 can happen automatically, so this is something better left for library to decide, and if we specify `force` what would that mean if server responds with a different version? cc @Lukasa @tanner0101 @ianpartridge

Reply:
I think taking the parameter out for now is a good idea. If we really need it, we can always add it back. And even if we end up needing it (we probably won't) we might decide to put it into the configuration... And as always I think it's better to start with the fewest options and add (& design) them once necessary.